src/bejeweled/board/Board.java

```diff
@@ -324,11 +326,11 @@ public void delete(Gem g, int newGem) {
 			}
 		}
 		else if(newGem == 1){
-			DoublePointsGem gem = new DoublePointsGem(g.row, g.col, g.type);
+			DoublePointsGem gem = (DoublePointsGem) gf.createGem(g.row, g.col, g.type, SpecialType.DOUBLE);
```

Comment:
You can just make it Gems here i guess, since DoublePointsGems are still Gems.
This avoids a cast.
Double check if it still works, but it should.